python/cugraph/structure/number_map.py

```diff
@@ -268,33 +303,41 @@ def compute(self):
                 # Compute the local partition id (obsolete once
                 #   https://github.com/dask/dask/issues/3707 is completed)
                 val_types["partition"] = np.int32
+
                 rehashed_with_partition_id = rehashed.map_partitions(
                     NumberMap.MultiGPU.compute_partition,
                     rehashed.divisions,
                     meta=val_types,
                 )
 
+                val_types.pop('count')
+
                 numbering_map = rehashed_with_partition_id.map_partitions(
-                    lambda df: df.groupby(self.col_names).min().reset_index()
+                    lambda df: df.groupby(
+                        self.col_names + ["hash", "partition"]
+                    ).sum()
+                    .sort_values('count', ascending=False)
```

Comment:
Do we have perf data associated with this change at this time? It would be helpful to add nightly MG performance regression tests. This would help to keep an eye on this so we (devs) don't have to track that manually.

Reply:
Opened an issue there https://github.com/rapidsai/cugraph/issues/1085

Reply:
We don't have any performance results at this time.

Reply:
Eventually, we need to implement fully optimized (most likely C++) version of this logic. Our python based approach can be a quick first step, but this doesn't look performance optimal.

Reply:
https://github.com/rapidsai/cugraph/issues/1048#issuecomment-672907741

and it will be great we can check this code can handle the common crawl data (possibly in a separate issue/PR) before closing #1048 .